ddtrace/tracer.py

```diff
@@ -258,7 +260,6 @@ def __init__(
         else:
             writer = AgentWriter(
                 agent_url=self._agent_url,
-                sampler=self._sampler,
```

Comment:
we missed passing the callback to this usage of the AgentWriter which caused the regression outlined in https://github.com/DataDog/dd-trace-py/pull/7333